src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -148,53 +148,70 @@ private static Double getGeneratorMarginalCost(Generator generator) {
         return null;
     }
 
-    private static List<Generator> computeAdjustableGenerators(Network network, Component component, List<String> generatorsWithFixedSupply,
-                                                               List<SubstationsGeneratorsOrderingInfos> substationsGeneratorsOrderingInfos,
-                                                               Reporter reporter) {
-        List<Generator> generatorsWithMarginalCost;
-
-        // get all generators in the component
-        List<Generator> generators = component.getBusStream().flatMap(Bus::getGeneratorStream).collect(Collectors.toList());
-
-        // remove generators with fixed supply
-        generators.removeIf(generator -> generatorsWithFixedSupply.contains(generator.getId()));
+    private static Map<Double, List<String>> getGeneratorsByMarginalCost(List<Generator> generators, Reporter reporter, String reporterSuffixKey) {
+        Map<Double, List<String>> generatorsByMarginalCost = new TreeMap<>();
 
         // set targetP to 0
         generators.forEach(generator -> generator.setTargetP(0.));
 
         // get generators with marginal cost
-        generatorsWithMarginalCost = generators.stream().filter(generator -> {
+        List<Generator> generatorsWithMarginalCost = generators.stream().filter(generator -> {
             Double marginalCost = getGeneratorMarginalCost(generator);
             if (marginalCost == null) {
-                report(reporter, Integer.toString(component.getNum()), "MissingMarginalCostForGenerator", "The generator ${generator} does not have a marginal cost",
-                    Map.of(GENERATOR, generator.getId()), TypedValue.WARN_SEVERITY);
+                report(reporter, reporterSuffixKey, "MissingMarginalCostForGenerator", "The generator ${generator} does not have a marginal cost",
+                        Map.of(GENERATOR, generator.getId()), TypedValue.TRACE_SEVERITY);
             }
             return marginalCost != null;
         }).collect(Collectors.toList());
 
+        int nbNoCost = generators.size() - generatorsWithMarginalCost.size();
+        if (nbNoCost > 0) {
```

Comment:
Add this synthetic report before each individual generator without marginal cost report ?